### ISSUE TITLE
Firehose with an ElasticSearch Destination is broken for those using ES versions 7 and above

### ISSUE THREAD

**Issue by chrisharrisonkiwi** [localstack/localstack#2544]
# Type of request: This is a ...

[x] bug report
[ ] feature request

Proposed labels:
[bug][diverges-from-aws][priority-high]

# Detailed description
There is a bug in the firehose API where it assumes an elasticsearch instance has a "TypeName"
Types have been removed in ES 7 and above
https://github.com/localstack/localstack/blob/master/localstack/services/firehose/firehose_api.py line 64 attempts to access a TypeName property which doesn't exist in ES 7 setups.

This results in:
```
File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 64, in put_records
es_type = es_dest['TypeName']
KeyError: 'TypeName'
```
When attempting a PutRecord action
...
## Client code 
https://github.com/localstack/localstack/blob/master/localstack/services/firehose/firehose_api.py line 64
...

**Comment by whummer** [comment]
Thanks for reporting this issue - great analysis @chrisharrisonkiwi . Would you mind creating a small pull request to fix this issue and make the Firehose API more resilient to different ES versions and the different response payloads. That would be much appreciated - thanks!

**Comment by whummer** [comment]
Should be fixed in #3024 - thanks again.
